data/state/accountsDB.go

```diff
@@ -123,18 +134,48 @@ func (adb *AccountsDB) LoadDataTrie(accountHandler AccountHandler) error {
 	return nil
 }
 
+// searches a data entry in journal from the most recent entries towards the first
+// and if it finds a matching data trie, returns a deep copy of that trie
+func (adb *AccountsDB) searchDataTrieInJournal(rootHash []byte) (data.Trie, error) {
+	adb.mutEntries.RLock()
+	defer adb.mutEntries.RUnlock()
+
+	for i := len(adb.entries) - 1; i >= 0; i-- {
+		dataTrieEntry, ok := adb.entries[i].(*BaseJournalEntryData)
+		if !ok {
+			continue
+		}
+
+		dataRootHash, err := dataTrieEntry.trie.Root()
+		if err != nil {
+			return nil, errors.New("corrupted trie found in journal while computing root hash: " + err.Error())
```

Comment:
do not make a new error. Add it into a constant one.

Reply:
+1

Reply:
Disagree, will be next to impossible to debug here the error generated. All you will see might be a "the node is nil" and since we also have a trie behind AccountState and n tries in accounts it will be hard to understand who output the error.